### ISSUE TITLE
Erroneously dispatching resources to an under-resourced cluster

### ISSUE THREAD

**Issue by whitewindmills** [karmada-io/karmada#2843]
**What happened**:
When I create a resource and dispatch it to a member cluster, it may be scheduled to a cluster with insufficient resources.
![image](https://user-images.githubusercontent.com/89241565/203218557-89ae8556-e134-4284-aa04-f6118c21d610.png)
![image](https://user-images.githubusercontent.com/89241565/203219226-1449b5e9-47c2-40e0-b9a2-be9771a30a23.png)

karmada-scheduler-estimator's log:
![image](https://user-images.githubusercontent.com/89241565/203221013-cf0652b2-45ba-40fb-9ad9-0b8798c8aefc.png)

And logically, this resource should not be scheduled to this member-57-85 cluster!

I do not find any code to filter the member cluster with zero replica,  and the karmada-scheduler treats zero as a legal value. I'm curious if this method `calAvailableReplicas ` should filter clusters with zero replica.
https://github.com/karmada-io/karmada/blob/master/pkg/scheduler/core/util.go#L35

**What you expected to happen**:
Resources should always be scheduled to a well-resourced member cluster.

**How to reproduce it (as minimally and precisely as possible)**:
Construct a member cluster with insufficient resources, and then distribute your resources.

**Anything else we need to know?**:

**Environment**:
- Karmada version: v1.3.0
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by Garrybest** [comment]
Is `Virtual Machine` a kind of workload? Actually, only workloads with resource and replicas will be tranmitted to `karmada-scheduler-estimator`. 

According to your logs, I think scheduler considers `Virtual Machine` as a workload and ask `karmada-scheduler-estimator` how many replicas the cluster is able to produce. As a result, 57-80 has 10 replicas available whereas 57-85 could not produce one neither. So the result will be 10 replicas for 57-80 and 0 replicas for 57-85.

**Comment by kingkonghong** [comment]
> Is `Virtual Machine` a kind of workload? Actually, only workloads with resource and replicas will be tranmitted to `karmada-scheduler-estimator`.
> 
> According to your logs, I think scheduler considers `Virtual Machine` as a workload and ask `karmada-scheduler-estimator` how many replicas the cluster is able to produce. As a result, 57-80 has 10 replicas available whereas 57-85 could not produce one neither. So the result will be 10 replicas for 57-80 and 0 replicas for 57-85.

You're right. But I don't think vm workloads should be scheduled to the `member-57-85` cluster. Do you have any suggestions?

**Comment by Garrybest** [comment]
Actually, it does schedule to this cluster but replicas is 0 now. So you think if the available replicas is 0, we should not propagate this workload to member cluster, right?

**Comment by kingkonghong** [comment]
> Actually, it does schedule to this cluster but replicas is 0 now. So you think if the available replicas is 0, we should not propagate this workload to member cluster, right?

Yes, I think the prerequisite for workloads to be scheduled is that there are sufficient resources.

**Comment by Garrybest** [comment]
+1, there is no need to propagate a cluster with no replicas assigned, we could check the result in assign function and remove all zero-replica cluster.

cc @RainbowMango for any ideas.

**Comment by XiShanYongYe-Chang** [comment]
> Yes, I think the prerequisite for workloads to be scheduled is that there are sufficient resources.

+1

 @kingkonghong Can you help submit a PR?

**Comment by kingkonghong** [comment]
> > Yes, I think the prerequisite for workloads to be scheduled is that there are sufficient resources.
> 
> +1
> 
> @kingkonghong Can you help submit a PR?

Sure, it's my pleasure!

**Comment by whitewindmills** [comment]
Perhaps there is no need for any modification.  First, for aggregation and dynamic weighting strategies, it is naturally no problem because it is divided by the number of replicas that can be assigned. Also for replication and static weight strategies, it should be a mandatory partitioning policy regardless of the number of allocable replicas of the cluster. I'll close this issue.

**Comment by XiShanYongYe-Chang** [comment]
Hi @whitewindmills, does this behavior meet your expectations?

**Comment by whitewindmills** [comment]
Not quite, actually, I haven't found a better way to resolve the conflict that comes with it. Certainly, I'd love to hear your good ideas.

**Comment by XiShanYongYe-Chang** [comment]
We can open this issue first and see if others have any solutions.
